### ISSUE TITLE
Crash report on 7.1: RejectedExecutionException in MediaGridAdapter.loadLocalImage

### ISSUE THREAD

**Issue by hypest** [wordpress-mobile/WordPress-Android#5668]
```
Fatal Exception: java.util.concurrent.RejectedExecutionException: Task android.os.AsyncTask$3@5439418 rejected from java.util.concurrent.ThreadPoolExecutor@7758e71[Running, pool size = 17, active threads = 17, queued tasks = 128, completed tasks = 15]
       at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2049)
       at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:814)
       at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1360)
       at android.os.AsyncTask.executeOnExecutor(AsyncTask.java:616)
       at org.wordpress.android.ui.media.MediaGridAdapter.loadLocalImage(MediaGridAdapter.java:376)
       at org.wordpress.android.ui.media.MediaGridAdapter.onBindViewHolder(MediaGridAdapter.java:137)
       at org.wordpress.android.ui.media.MediaGridAdapter.onBindViewHolder(MediaGridAdapter.java:43)
       at android.support.v7.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:6308)
       at android.support.v7.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:6341)
       at android.support.v7.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:5287)
       at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5550)
       at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5392)
       at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5388)
       at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2149)
       at android.support.v7.widget.GridLayoutManager.layoutChunk(GridLayoutManager.java:556)
       at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1496)
       at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:593)
       at android.support.v7.widget.GridLayoutManager.onLayoutChildren(GridLayoutManager.java:170)
       at android.support.v7.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3535)
       at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:3264)
       at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:3796)
       at android.view.View.layout(View.java:18793)
       at android.view.ViewGroup.layout(ViewGroup.java:5952)
       at android.support.v4.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:636)
       at android.view.View.layout(View.java:18793)
       at android.view.ViewGroup.layout(ViewGroup.java:5952)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
       at android.view.View.layout(View.java:18793)
       at android.view.ViewGroup.layout(ViewGroup.java:5952)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at android.view.View.layout(View.java:18793)
       at android.view.ViewGroup.layout(ViewGroup.java:5952)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
       at android.view.View.layout(View.java:18793)
       at android.view.ViewGroup.layout(ViewGroup.java:5952)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
       at android.view.View.layout(View.java:18793)
       at android.view.ViewGroup.layout(ViewGroup.java:5952)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at android.view.View.layout(View.java:18793)
       at android.view.ViewGroup.layout(ViewGroup.java:5952)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at android.view.View.layout(View.java:18793)
       at android.view.ViewGroup.layout(ViewGroup.java:5952)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at android.view.View.layout(View.java:18793)
       at android.view.ViewGroup.layout(ViewGroup.java:5952)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
       at android.view.View.layout(View.java:18793)
       at android.view.ViewGroup.layout(ViewGroup.java:5952)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at com.android.internal.policy.DecorView.onLayout(DecorView.java:818)
       at android.view.View.layout(View.java:18793)
       at android.view.ViewGroup.layout(ViewGroup.java:5952)
       at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2615)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2331)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1490)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7027)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:927)
       at android.view.Choreographer.doCallbacks(Choreographer.java:702)
       at android.view.Choreographer.doFrame(Choreographer.java:638)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:913)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6692)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
```

**Comment by hypest** [comment]
Probably related to low free resources (or high resource demand, depends how you look at it) but this also happens:
```
Fatal Exception: java.util.concurrent.RejectedExecutionException: Task android.os.AsyncTask$3@42786010 rejected from java.util.concurrent.ThreadPoolExecutor@41c46fc8[Running, pool size = 9, active threads = 9, queued tasks = 128, completed tasks = 65]
       at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2011)
       at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:793)
       at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1339)
       at android.os.AsyncTask.executeOnExecutor(AsyncTask.java:590)
       at com.android.internal.app.ResolverActivity$ResolveListAdapter.bindView(ResolverActivity.java:768)
       at com.android.internal.app.ResolverActivity$ResolveListAdapter.getView(ResolverActivity.java:752)
       at android.widget.AbsListView.obtainView(AbsListView.java:2250)
       at android.widget.ListView.measureHeightOfChildren(ListView.java:1271)
       at android.widget.ListView.onMeasure(ListView.java:1183)
       at android.view.View.measure(View.java:16512)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5134)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
       at android.view.View.measure(View.java:16512)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5134)
       at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1404)
       at android.widget.LinearLayout.measureVertical(LinearLayout.java:695)
       at android.widget.LinearLayout.onMeasure(LinearLayout.java:588)
       at android.view.View.measure(View.java:16512)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5134)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
       at android.view.View.measure(View.java:16512)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5134)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
       at android.view.View.measure(View.java:16512)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5134)
       at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1404)
       at android.widget.LinearLayout.measureVertical(LinearLayout.java:695)
       at android.widget.LinearLayout.onMeasure(LinearLayout.java:588)
       at android.view.View.measure(View.java:16512)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5134)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
       at android.view.View.measure(View.java:16512)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5134)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
       at android.view.View.measure(View.java:16512)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5134)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
       at com.android.internal.policy.impl.PhoneWindow$DecorView.onMeasure(PhoneWindow.java:2332)
       at android.view.View.measure(View.java:16512)
       at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:1921)
       at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1113)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1295)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1000)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5735)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:761)
       at android.view.Choreographer.doCallbacks(Choreographer.java:574)
       at android.view.Choreographer.doFrame(Choreographer.java:544)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:747)
       at android.os.Handler.handleCallback(Handler.java:733)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5387)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:974)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:790)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

**Comment by daniloercoli** [comment]
cc @nbradbury

**Comment by daniloercoli** [comment]
Fixed in #5722
